src/tls.rs

```diff
@@ -146,6 +168,22 @@ impl TlsSettings {
             identity,
         })
     }
+
+    pub fn identity(&self) -> Option<Identity> {
+        // This data was test-built previously, so we can just use it
+        // here and expect the results will not fail. This can all be
+        // reworked when `native_tls::Identity` gains the Clone impl.
+        self.identity.as_ref().map(|identity| {
+            Identity::from_pkcs12(&identity.0, &identity.1).expect("Could not build identity")
+        })
+    }
+
+    pub(crate) fn acceptor(&self) -> crate::Result<TlsAcceptor> {
+        match self.identity() {
```

Comment:
minor nit: this would be better if it were just an if let imo

Reply:
I think this is the `match` vs `if` debate again... (ref #1720)

Reply:
Though I think the reason I prefer if let here is because the option enumerates to exactly two options here therefore we don't need the extra syntax for matching None. I don't mind either way.